libasynql/src/poggit/libasynql/base/QuerySendQueue.php

```diff
@@ -23,21 +23,48 @@
 namespace poggit\libasynql\base;
 
 use Threaded;
-use function is_string;
 use function serialize;
-use function unserialize;
 
 class QuerySendQueue extends Threaded{
-	public function scheduleQuery(int $queryId, int $mode, string $query, array $params) : void{
-		$this[] = serialize([$queryId, $mode, $query, $params]);
+	/** @var bool */
+	private $invalidated = false;
+	/** @var Threaded */
+	private $queries;
+
+	public function __construct(){
+		$this->queries = new Threaded();
```

Comment:
Is there a particular reason not to use `$this`?

Reply:
it required an encapsulated threaded queue so that we could shift ***that*** instead of the main one which now has an "invalidate" property... for some unknown reason, shifting a pthreads `Threaded` object shifts its properties too.